lib/spack/spack/solver/asp.py

```diff
@@ -277,7 +278,16 @@ def __init__(self, cores=True, asp=None):
             asp (file-like): optional stream to write a text-based ASP program
                 for debugging or verification.
         """
-        assert clingo, "PyclingoDriver requires clingo with Python support"
+        global clingo
+        if not clingo:
+            # TODO: Find a way to vendor the concrete spec
+            # in a cross-platform way
+            clingo_spec = spack.spec.Spec('clingo@spack+python')
```

Comment:
Once #20652 is in we can modify this to:
```suggestion
            generic_target = archspec.cpu.host().family
            spec_str = 'clingo-bootstrap@spack+python target={0}'.format(
                str(generic_target)
            )
            clingo_spec = spack.spec.Spec(spec_str)
```